core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/SplittingTransformer.java

```diff
@@ -259,28 +268,27 @@ private void handleFutureCancel(Throwable e) {
      * body publisher.
      */
     private class IndividualTransformer implements AsyncResponseTransformer<ResponseT, ResponseT> {
+        private final boolean isFirstPart;
```

Comment:
Can we use partNumber instead? We will need to use it when we start to work on adding more validations